src/rust/engine/fs/store/benches/store.rs

```diff
@@ -75,6 +76,54 @@ pub fn criterion_benchmark_materialize(c: &mut Criterion) {
   }
 }
 
+///
+/// NB: More accurately, this benchmarks `Snapshot::digest_from_path_stats`, which avoids
+/// filesystem traversal overheads and focuses on digesting/capturing.
+///
+pub fn criterion_benchmark_snapshot_capture(c: &mut Criterion) {
+  let executor = Executor::global(num_cpus::get(), num_cpus::get() * 4).unwrap();
+
+  let mut cgroup = c.benchmark_group("snapshot_capture");
+
+  // The number of files, file size, whether the inputs should be assumed to be immutable, and the
+  // number of times to capture (only the first capture actually stores anything: the rest should
+  // ignore the duplicated data.)
+  for params in vec![
+    (100, 100, false, 100),
+    (20, 10_000_000, true, 10),
+    (1, 200_000_000, true, 10),
+  ] {
+    let (count, size, _immutable, captures) = params;
```

Comment:
The `_immutable` flag becomes relevant in #12563.

Reply:
I'm not sure I understand the point of `captures > 1`? If we expect them to be no-ops, why are we benchmarking them?

Reply:
Anytime we `Snapshot` the filesystem, if we discover that we've already captured a file in the LMDB store, we'll skip re-storing it. But that still involves hashing it to determine that we don't need to store it again. So it's a noop in terms of storage, but not in terms of hashing: `LMDB::put` will skip storing the value, and in #12563, `LMDB::reserve` will fail with an "exists" error and prevent the second pass over the data.

The captures ratio here is supposed to align with usecases (I'll add a comment in #12563):
* `(100, 100, false, 100)` is supposed to represent Snapshotting the workspace: lots of small hand written files that change relatively infrequently (100 captures is 1 write and 99 noops)
* `(20, 10_000_000, true, 10)` and `(1, 200_000_000, true, 10)` both represent capturing out of local process execution sandboxes: immutable data that is more unique, so more writes and fewer noops.